aten/src/ATen/native/sparse/utils/ParamUtils.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/TensorUtils.h>
+#include <tuple>
+
+namespace at {
+namespace native {
+
+namespace apply {
+
+std::pair<Tensor, Tensor> softmax_sparse_check_invariants(
+    const Tensor& input_,
+    const int64_t dim_,
+    const bool half_to_float,
+    CheckedFrom function_name) {
+  TORCH_INTERNAL_ASSERT(input_.is_sparse());
+  TORCH_CHECK(
+      !half_to_float,
+      std::string(function_name) +
+          ": with half to float conversion is not supported on CPU");
+  auto input = input_.coalesce();
```

Comment:
It's strange that the input is coalesced and the output constructed in a function called "check_invariants"

Reply:
(seems like a function named "check_invariants" should return void or bool)

Reply:
The easiest fix may be renaming this function (and the next one)